homeassistant/components/esphome/manager.py

```diff
@@ -378,6 +391,24 @@ def _async_on_log(self, msg: SubscribeLogsResponse) -> None:
                 ANSI_ESCAPE_78BIT.sub(b"", log).decode("utf-8", "backslashreplace"),
             )
 
+    @callback
+    def _async_get_equivalent_log_level(self) -> LogLevel:
+        """Get the equivalent ESPHome log level for the current logger."""
+        return LOGGER_TO_LOG_LEVEL.get(
+            _LOGGER.getEffectiveLevel(), LogLevel.LOG_LEVEL_VERY_VERBOSE
+        )
+
+    @callback
+    def _async_subscribe_logs(self, log_level: LogLevel) -> None:
+        """Subscribe to logs."""
+        if self._cancel_subscribe_logs is not None:
+            self._cancel_subscribe_logs()
+            self._cancel_subscribe_logs = None
+        self._log_level = log_level
+        self._cancel_subscribe_logs = self.cli.subscribe_logs(
```

Comment:
It's intentional that we don't call this on disconnect since it automatically gets cleaned up. We only need to call it if we change the log level on the remote to adjust the callback subscription